res/controllers/Traktor-Kontrol-S2-MK2-hid-scripts.js

```diff
@@ -0,0 +1,1389 @@
+  MessageShort.setCallback("[Channel1]", "!play", this.play);
+  MessageShort.setCallback("[Channel2]", "!play", this.play);
+  MessageShort.setCallback("[Channel1]", "!pad1", this.pad);
+  MessageShort.setCallback("[Channel1]", "!pad2", this.pad);
+  MessageShort.setCallback("[Channel1]", "!pad3", this.pad);
+  MessageShort.setCallback("[Channel1]", "!pad4", this.pad);
+  MessageShort.setCallback("[Channel1]", "!remix_button", this.samplerModeButton);
+  MessageShort.setCallback("[Channel1]", "!flux_button", this.introOutroModeButton);
+  MessageShort.setCallback("[Channel2]", "!pad1", this.pad);
+  MessageShort.setCallback("[Channel2]", "!pad2", this.pad);
+  MessageShort.setCallback("[Channel2]", "!pad3", this.pad);
+  MessageShort.setCallback("[Channel2]", "!pad4", this.pad);
+  MessageShort.setCallback("[Channel2]", "!remix_button", this.samplerModeButton);
+  MessageShort.setCallback("[Channel2]", "!flux_button", this.introOutroModeButton);
+  MessageShort.setCallback("[Channel1]", "!loop_in", this.loopIn);
+  MessageShort.setCallback("[Channel1]", "!loop_out", this.loopOut);
+  MessageShort.setCallback("[Channel2]", "!loop_in", this.loopIn);
+  MessageShort.setCallback("[Channel2]", "!loop_out", this.loopOut);
+  MessageShort.setCallback("[Channel1]", "!load_track", this.loadTrack);
+  MessageShort.setCallback("[Channel2]", "!load_track", this.loadTrack);
+  MessageShort.setCallback("[Channel1]", "!sync_enabled", this.syncEnabled);
+  MessageShort.setCallback("[Channel2]", "!sync_enabled", this.syncEnabled);
+  MessageShort.setCallback("[Channel1]", "!left_encoder_press", this.leftEncoderPress);
+  MessageShort.setCallback("[Channel2]", "!left_encoder_press", this.leftEncoderPress);
+  MessageShort.setCallback("[Channel1]", "!right_encoder_press", this.rightEncoderPress);
+  MessageShort.setCallback("[Channel2]", "!right_encoder_press", this.rightEncoderPress);
+  MessageShort.setCallback("[Channel1]", "!jog_touch", this.jogTouch);
+  MessageShort.setCallback("[Channel2]", "!jog_touch", this.jogTouch);
+  MessageShort.setCallback("[Channel1]", "!jog_wheel", this.jogMove);
+  MessageShort.setCallback("[Channel2]", "!jog_wheel", this.jogMove);
+  
+  MessageShort.setCallback("[EffectRack1_EffectUnit1]", "!effect_focus_button", this.effectFocusButton);
+  MessageShort.setCallback("[EffectRack1_EffectUnit1", "!effectbutton1", this.effectButton);
+  MessageShort.setCallback("[EffectRack1_EffectUnit1", "!effectbutton2", this.effectButton);
+  MessageShort.setCallback("[EffectRack1_EffectUnit1", "!effectbutton3", this.effectButton);
+ 
+  MessageShort.setCallback("[EffectRack1_EffectUnit2]", "!effect_focus_button", this.effectFocusButton);
+  MessageShort.setCallback("[EffectRack1_EffectUnit2", "!effectbutton1", this.effectButton);
+  MessageShort.setCallback("[EffectRack1_EffectUnit2", "!effectbutton2", this.effectButton);
+  MessageShort.setCallback("[EffectRack1_EffectUnit2", "!effectbutton3", this.effectButton);
```

Comment:
Can you condense this sections by constructing this in a loop?
eg 
```javascript
for (group of ["[Channel1]", "[Channel2]"]) {
      MessageShort.setCallback(group, "!top_encoder_press", this.topEncoderPress);
      ...
}
```

Reply:
Better yet, I made these `setCallback` calls unnecessary by letting the script provide the callback to `addControl`.